src/rust/engine/watch/src/lib.rs

```diff
@@ -197,6 +197,20 @@ impl InvalidationWatcher {
     canonical_build_root: &Path,
     ev: Event,
   ) {
+    if matches!(ev.kind, EventKind::Modify(ModifyKind::Metadata(_)))
+      && !matches!(
+        ev.kind,
+        EventKind::Modify(ModifyKind::Metadata(MetadataKind::Permissions))
+      )
+    {
+      // If only the metadata (AccessTime, WriteTime, Perms, Link Count, etc...) was changed
+      // (other than permissions, which include the executable bit) it doesn't change anything Pants
+      // particularly cares about. One could argue if the permissions/ownership changed Pants would
```

Comment:
> One could argue if the permissions/ownership changed Pants would care ...

That wording seems off since your guard clause _does_ care about `MetadataKind::Permissions`.